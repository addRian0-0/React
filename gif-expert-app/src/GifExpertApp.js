import React, { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

export const GifExpertApp = ({ defaultCategories = [] }) => {
  
  //const [categorias, setcategorias] = useState(['Arcane']);
  const [categorias, setcategorias] = useState(defaultCategories);

  return (
    <>
      <h2>GifExpertApp</h2>
      <hr/>
      <AddCategory setCategories={setcategorias} />
      <ul>
        {
          categorias.map((c, i) => {
            return <GifGrid key={i} category={c} />
          })
        }
      </ul>
    </>
  );
};

export default GifExpertApp;
