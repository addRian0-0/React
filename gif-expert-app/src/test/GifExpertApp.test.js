import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('pruebas en el componente principal de la aplicacion', () => {

    const wrapper = shallow( <GifExpertApp/> )

    test('snapshot al componente', () => {
        
        expect(wrapper).toMatchSnapshot();

    })

    test('mostrar una lista de categroias', () => {
        
        const categorias = ["Arcane", "Goku"];
        const wrapper = shallow( <GifExpertApp defaultCategories={categorias} /> )

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categorias.length);

    })
    

});