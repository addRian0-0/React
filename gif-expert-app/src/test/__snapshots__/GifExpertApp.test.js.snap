// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente principal de la aplicacion mostrar una lista de categroias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ul>
    <GifGrid
      category="Arcane"
      key="0"
    />
    <GifGrid
      category="Goku"
      key="1"
    />
  </ul>
</Fragment>
`;

exports[`pruebas en el componente principal de la aplicacion snapshot al componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ul />
</Fragment>
`;
